src/rebar3_hex_publish.erl

```diff
@@ -339,12 +331,87 @@ maybe_prompt(_Args, Message) ->
             abort
     end.
 
-create_tarball(#{metadata := Meta, files := Files}) ->
-    case hex_tarball:create(Meta, Files) of
-        {ok, #{tarball := Tarball, inner_checksum := _Checksum}} ->
-            Tarball;
-        Error ->
-            ?RAISE(Error)
+hex_opts(Opts) ->
+    lists:filter(fun({replace, _}) -> true;
+                    ({_,_}) -> false 
+                 end,
+                 maps:to_list(Opts)).
+
+%%% ===================================================================
+%%% doc publishing and related functions
+%%% ===================================================================
+
+publish_docs(State, Repo, App, Args) ->
+    #{tarball := Tar, name := Name, vsn := Vsn} = create_docs(State, Repo, App),
+    case Args of
+        #{dry_run := true} ->
+            rebar_api:info("--dry-run enabled : will not publish docs.", []),
+            {ok, State};
+         _ ->
+            case rebar3_hex_client:publish_docs(Repo, Name, Vsn, Tar) of
+                {ok, _} ->
+                    rebar_api:info("Published docs for ~ts ~ts", [Name, Vsn]),
+                    {ok, State};
+                Reason ->
+                    ?RAISE({publish_docs, Reason})
+            end
+    end.
+
+create_docs(State, Repo, App) -> 
+    case rebar3_hex_build:create_docs(State, Repo, App) of 
+        {ok, Docs} ->
+            Docs;
+        Err -> 
+            ?RAISE({create_docs, Err})
+    end.
+
+%%% ===================================================================
+%%% package and doc reversion functions
+%%% ===================================================================
+
+revert_package(State, Repo, AppName, Vsn) ->
+    BinAppName = rebar_utils:to_binary(AppName),
+    BinVsn =  rebar_utils:to_binary(Vsn),
+    assert_valid_version_arg(BinVsn),
+    {ok, HexConfig} = write_config(Repo),
+    case rebar3_hex_client:delete_release(HexConfig, BinAppName, BinVsn) of
+        {ok, _} -> 
+            rebar_api:info("Successfully deleted package ~ts ~ts", [AppName, Vsn]),
+            Prompt = io_lib:format("Also delete tag v~s?", [Vsn]),
```

Comment:
`~ts` here too? Even with semver, trailing parts can be added as labels and I don't recall if there's a strict need for it to be unicode-safe.